attributes/default.rb

```diff
@@ -23,3 +23,4 @@
 default['varnish']['VARNISH_WORKING_DIR'] = ''
 default['varnish']['GeoIP_enabled'] = false
 default['varnish']['release_rpm'] = 'http://repo.varnish-cache.org/redhat/varnish-3.0/el5/noarch/varnish-release/varnish-release-3.0-1.noarch.rpm'
+default['varnish']['custom_parameters'] = []
```

Comment:
This should be "{}" by default, as you are expecting a hash rather than an array